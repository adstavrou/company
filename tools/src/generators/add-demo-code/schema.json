{
  "$schema": "http://json-schema.org/schema",
  "title": "Add Demo Generator",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The folder name for the new demo (e.g. buttons-demo)"
    },
    "title": {
      "type": "string",
      "description": "The display title for the demo (e.g. Buttons Demo)"
    }
  }
}