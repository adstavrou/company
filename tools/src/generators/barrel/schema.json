{
    "$schema": "http://json-schema.org/schema",
    "title": "Barrel Generator",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "The name of the project (library) to generate the barrel for"
        }
    },
    "required": [
        "project"
    ]
}