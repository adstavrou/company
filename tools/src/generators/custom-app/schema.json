{
  "$schema": "http://json-schema.org/schema",
  "title": "Custom App Generator",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the app"
    }
  }
}