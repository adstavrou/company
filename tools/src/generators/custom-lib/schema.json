{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Lib Generator",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Library name"
        },
        "build": {
            "type": "boolean",
            "description": "Add build target?"
        },
        "unitTest": {
            "type": "boolean",
            "description": "Add unit tests?"
        }
    },
    "required": []
}