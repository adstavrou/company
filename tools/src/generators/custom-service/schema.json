{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Service Generator",
    "type": "object",
    "properties": {
      "name": {
        "type": "string",
        "description": "Name of the service"
      },
      "directory": {
        "type": "string",
        "description": "Target directory inside libs/"
      }
    },
    "required": []
  }