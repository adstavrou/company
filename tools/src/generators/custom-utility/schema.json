{
    "$schema": "http://json-schema.org/schema",
    "title": "Custom Utility Generator",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "The name of the utility function"
        },
        "directory": {
            "type": "string",
            "description": "The library folder (e.g. shared, core-utils)"
        },
        "withTest": {
            "type": "boolean",
            "description": "Generate test file"
        }
    },
    "required": []
}