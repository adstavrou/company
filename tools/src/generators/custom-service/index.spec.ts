import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';
import generator from './index';
import { CustomServiceGeneratorSchema } from './schema';

describe('custom-service generator (snapshot)', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
  });

  it('should generate service file and match snapshot', async () => {
    const options: CustomServiceGeneratorSchema = {
      name: 'auth manager',
      directory: 'shared',
    };

    await generator(tree, options);

    const filePath = 'libs/shared/src/auth-manager.service.ts';
    const content = tree.read(filePath, 'utf-8');

    expect(content).toMatchSnapshot();
  });
});
