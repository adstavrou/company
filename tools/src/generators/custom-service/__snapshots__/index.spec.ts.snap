// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-service generator (snapshot) should generate service file and match snapshot 1`] = `
"export class AuthManagerService {
  constructor() {
    console.log('AuthManagerService created!');
  }
}
"
`;
